src/allmydata/crypto/rsa.py

```diff
@@ -0,0 +1,156 @@
+from cryptography.exceptions import InvalidSignature
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import rsa, padding
+from cryptography.hazmat.primitives.serialization import load_der_private_key, load_der_public_key, \
+    Encoding, PrivateFormat, PublicFormat, NoEncryption
+
+from allmydata.crypto import BadSignature
+
+
+"""
+This is the value that was used by `pycryptopp`, and we must continue to use it for
+both backwards compatibility and interoperability.
+
+The docs for `cryptography` suggest to use the constant defined at
+`cryptography.hazmat.primitives.asymmetric.padding.PSS.MAX_LENGTH`, but this causes old
+signatures to fail to validate.
+"""
+RSA_PSS_SALT_LENGTH = 32
+
+
+def create_signing_keypair(key_size):
+    """
+    Create a new RSA signing (private) keypair from scratch. Can be used with
+    `sign_data` function.
+
+    :param int key_size: length of key in bits
+
+    :returns: 2-tuple of (private_key, public_key)
+    """
+    # as per
+    # https://cryptography.io/en/latest/hazmat/primitives/asymmetric/rsa/#cryptography.hazmat.primitives.asymmetric.rsa.generate_private_key
+    # and
+    # https://www.daemonology.net/blog/2009-06-11-cryptographic-right-answers.htmlhttps://www.daemonology.net/blog/2009-06-11-cryptographic-right-answers.html
+    # the public exponent is 65537
+    # (I *believe* that pycryptopp would have used cryptopp's default,
+    # though, which is 17)
+    priv_key = rsa.generate_private_key(
+        public_exponent=65537,
```

Comment:
https://security.stackexchange.com/questions/2335/should-rsa-public-exponent-be-only-in-3-5-17-257-or-65537-due-to-security-c has some answers explaining some of what goes into exponent choice.

From my reading there, I might lean towards 17.  We might try to measure the performance consequences of changing to 65537 before doing so.  Also, everything I read suggests 17 provides all of the same security properties as 65537 (assuming correct padding implementation and private key generation).

Reply:
Okay, maybe we should use `17` for now then, and include this link (and slightly more discussion, i.e. that pycryptopp / cryptopp was using `17` all along)

Reply:
Okay, I have added the above link and also flipped back to use 17 -- if only because we're only trying to change to cryptography library, not change defaults?